From 66369761c9e8304ce0d9690cfafb46b4f4d2d6cc Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Thu, 26 Jul 2012 09:52:45 +0800
Subject: [PATCH 1815/2463] ENGR00218412-1 OV5640 mipi:Power down after
 checking dev id

This patch powers down camera after checking camera device
id to save power.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
(cherry picked from commit a73b70b3d425825b4f4ba99c4c38c23bde227a9a)
---
 drivers/media/video/mxc/capture/ov5640_mipi.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/media/video/mxc/capture/ov5640_mipi.c b/drivers/media/video/mxc/capture/ov5640_mipi.c
index b46820b..f19fdc3 100644
--- a/drivers/media/video/mxc/capture/ov5640_mipi.c
+++ b/drivers/media/video/mxc/capture/ov5640_mipi.c
@@ -1871,6 +1871,9 @@ static int ov5640_probe(struct i2c_client *client,
 		goto err4;
 	}
 
+	if (plat_data->pwdn)
+		plat_data->pwdn(1);
+
 	camera_plat = plat_data;
 
 	ov5640_int_device.priv = &ov5640_data;
-- 
1.7.7.4

